analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ProgramState.cs

```diff
@@ -125,10 +125,20 @@ public ProgramState ResetOperations()
 
     public IOperation ResolveCapture(IOperation operation) =>
         operation?.Kind == OperationKindEx.FlowCaptureReference
-        && this[IFlowCaptureReferenceOperationWrapper.FromOperation(operation).Id] is { } captured
+        && this[operation.ToFlowCaptureReference().Id] is { } captured
             ? captured
             : operation;
 
+    public IOperation ResolveCaptureAndUnwrapConversion(IOperation operation) =>
+        operation?.Kind switch
```

Comment:
This should never be called for `null`. And recursion will never cause it
```suggestion
        operation.Kind switch
```

Reply:
This was to be consistent with `ResolveCapture`.